week3/Q1.py

```diff
@@ -0,0 +1,177 @@
+def get_graph(values, n, m):
+	'''
+	Builds a graph.
+
+	Args:
+		values: str, elements of the table, all rows in one line
+		n: int, number of columns
+		m: int, number of rows
+
+	Returns:
+		dict, includes two elements with keys 'nodes' (stores a dict where key is number of a node and value is node's value) and 'vertices' (stores a dict where key is a number of a node and value is a set of numbers of nodes to which it is connected)
+	'''
+	nodes = {i: values[i] for i in range(n*m)}
+	vertices = []
+	for i in range(n*m):
+		if n == 1 and m == 1:
+			vertices.append({})
+			break
+		elif n == 1 or m == 1:
+			if i == 0:
+				vertices.append({i + 1})
+			elif (i == m - 1 and n == 1) or (i == n - 1 and m == 1):
+				vertices.append({i - 1})
+			else:
+				vertices.append({i - 1, i + 1})
+		else:
+			#up left corner
```

Comment:
Suggestion: You can probably make this a bit shorted if you use a `for`-loop (that is aware of the boundary) to generate the list of adjacent cells.